extensions/security-webauthn/runtime/src/main/java/io/quarkus/security/webauthn/WebAuthnAuthenticationMechanism.java

```diff
@@ -0,0 +1,81 @@
+package io.quarkus.security.webauthn;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.request.AuthenticationRequest;
+import io.quarkus.security.identity.request.TrustedAuthenticationRequest;
+import io.quarkus.vertx.http.runtime.security.ChallengeData;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
+import io.quarkus.vertx.http.runtime.security.HttpCredentialTransport;
+import io.quarkus.vertx.http.runtime.security.HttpSecurityUtils;
+import io.quarkus.vertx.http.runtime.security.PersistentLoginManager;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * An AuthenticationMechanism for WebAuthn which mostly delegates to @{link PersistentLoginManager}
+ * and @{TrustedAuthenticationRequest}, since authentication is handled by {@link WebAuthnController}
+ * or @{link WebAuthnSecurity}.
+ */
+public class WebAuthnAuthenticationMechanism implements HttpAuthenticationMechanism {
+
+    private static final Logger log = Logger.getLogger(WebAuthnAuthenticationMechanism.class);
+
+    private PersistentLoginManager loginManager;
+    private String loginPage;
+
+    WebAuthnAuthenticationMechanism(PersistentLoginManager loginManager, String loginPage) {
+        this.loginManager = loginManager;
+        this.loginPage = loginPage;
+    }
+
+    @Override
+    public Uni<SecurityIdentity> authenticate(RoutingContext context, IdentityProviderManager identityProviderManager) {
+        PersistentLoginManager.RestoreResult result = loginManager.restore(context);
+        if (result != null) {
+            context.put(HttpAuthenticationMechanism.class.getName(), this);
+            Uni<SecurityIdentity> ret = identityProviderManager
+                    .authenticate(HttpSecurityUtils
+                            .setRoutingContextAttribute(new TrustedAuthenticationRequest(result.getPrincipal()), context));
```

Comment:
I wonder if `WebAuthn` specific request type should be introduced, as `Trusted` one might be only applicable to the MTLS mechanism, @stuartwdouglas, what do you think ?

Reply:
Actually, it is already available, so I think it should be used instead, and this mechanism should not advertize it can support a `Trusted` one

Reply:
AFAIK this is actually only used for the cookie session stored after Form-based auth. Not MTLS.

Reply:
OK, these are mostly the internal details for now so I guess we can change things if needed later

Reply:
Cool